tests/databases/value/mongo_db/test_value_update.py

```diff
@@ -0,0 +1,233 @@
+import pymongo
+
+from .utilities import (
+    get_database_state,
+)
+from ...utilities import (
+    DatabaseEntry,
+    DatabaseState,
+    assert_database_state,
+)
+
+
+def test_update_1():
+    """
+    Test that existing entries are updated.
+
+    """
+
+    collection = '_test_update_1'
+    database_name = '_test_update_1'
+    client = pymongo.MongoClient()
+    client.drop_database(database_name)
+
+    database = stk.ValueMongoDb(
+        mongo_client=client,
+        collection=collection,
+        database=database_name,
+        put_lru_cache_size=0,
+        get_lru_cache_size=0,
+    )
+
+    molecule = stk.BuildingBlock('CCC')
+
+    database.put(molecule, 12)
+    assert_database_state(
+        state1=get_database_state(database),
+        state2=DatabaseState({
+            DatabaseEntry(
+                InChIKey=stk.InchiKey().get_key(molecule),
+                v=12,
+            ): 1,
+        }),
+    )
+
+    database.put(molecule, 43)
+    assert_database_state(
+        state1=get_database_state(database),
+        state2=DatabaseState({
+            DatabaseEntry(
+                InChIKey=stk.InchiKey().get_key(molecule),
+                v=43,
+            ): 1,
+        }),
+    )
+
+
+def test_update_2():
+    """
+    Test that existing entries are updated.
+
+    In this test, your first create two separate entries, using
+    different molecule keys. You then update both at the same time,
+    with a database which uses both molecule keys.
+
+    """
+
+    collection = '_test_update_2'
+    database_name = '_test_update_2'
+    client = pymongo.MongoClient()
+    client.drop_database(database_name)
+
+    database1 = stk.ValueMongoDb(
+        mongo_client=client,
+        collection=collection,
+        database=database_name,
+        put_lru_cache_size=0,
+        get_lru_cache_size=0,
+        key_makers=(
+            stk.InchiKey(),
+        ),
+    )
+    database2 = stk.ValueMongoDb(
+        mongo_client=client,
+        collection=collection,
+        database=database_name,
+        put_lru_cache_size=0,
+        get_lru_cache_size=0,
+        key_makers=(
+            stk.Smiles(),
+        ),
+    )
+    database3 = stk.ValueMongoDb(
+        mongo_client=client,
+        collection=collection,
+        database=database_name,
+        put_lru_cache_size=0,
+        get_lru_cache_size=0,
+        key_makers=(
+            stk.InchiKey(),
+            stk.Smiles(),
+        ),
+    )
+
+    molecule = stk.BuildingBlock('CCC')
+
+    database1.put(molecule, 12)
+    assert_database_state(
+        state1=get_database_state(database1),
+        state2=DatabaseState({
+            DatabaseEntry(
+                InChIKey=stk.InchiKey().get_key(molecule),
+                v=12,
+            ): 1,
+        }),
+    )
+
+    # Should add another entry, as a different key maker is used.
+    database2.put(molecule, 32)
+    assert_database_state(
+        state1=get_database_state(database1),
```

Comment:
should this not be database2? If not, then I am confused what this test is doing and the docs should be updated.

Reply:
I think I get it now, they are the same database. But your docs should be more clear.

Reply:
Any more specific suggestions?